### PR TITLE
Allow user to set y size of belt gap.

### PR DESCRIPTION
**Pull request by martinbudden**
Fixes https://github.com/nophead/NopSCADlib/issues/91

### CODE REVIEW

File: vitamins/belt.scad [hunk 1]
```diff
@@ -64,8 +64,8 @@ module belt(type, points, gap = 0, gap_pos = undef, belt_colour = grey(20), toot
         if(gap)
             translate([gap_pos.x, gap_pos.y])
                 rotate(is_undef(gap_pos.z) ? 0 : gap_pos.z)
-                    translate([0, ph - thickness / 2])
-                        square([gap, thickness + eps], center = true);
+                    translate([0, is_list(gap) ? -th / 4 : ph - thickness / 2])
```
Comment (nophead) [inline vitamins/belt.scad:67]: Why tooth height / 4?

Comment (martinbudden) [reply vitamins/belt.scad:67]: That centers it on the gap.

Comment (nophead) [reply vitamins/belt.scad:67]: I don't see how. The belt is modelled relative to the pitch line, so adding the pitch height and subtracting half the thickness gets to the middle of the belt, so a rectangle slight thicker thicker than the belt just covers it.

How is the tooth height relevant and why divide it by 4?

Comment (martinbudden) [reply vitamins/belt.scad:67]: I'll look into it again.

Comment (nophead) [reply vitamins/belt.scad:67]: If gap.y is the offset from one end to the other that I think you need to add the belt thickness + eps to that to cover both. And the centre would be ph - thickness / 2 + gap.y / 2, I think.

Comment (martinbudden) [reply vitamins/belt.scad:67]: OK, I think it should be:
```
                    translate([0, (ph - thickness) / 2]);
```
ie, I think your original value of 
```
                    translate([0, ph - thickness / 2]);
```
is incorrect.

But in any case, the value does not need to change when a y value for the gap is set. ie the full code for `gap` is
```
    module gap()
        if(gap)
            translate([gap_pos.x, gap_pos.y])
                rotate(is_undef(gap_pos.z) ? 0 : gap_pos.z)
                    translate([0, (ph - thickness) / 2])
                        square(is_list(gap) ? [gap.x, gap.y]: [gap, thickness + eps], center = true);
```
But in any case, the difference is marginal, so I won't argue against it staying as `ph - thinkness / 2`.

Comment (nophead) [reply vitamins/belt.scad:67]: Why would you divide ph by 2? The pitch height is a magic number part way through the teeth measured from the tips of the teeth. Adding that gets to the tips and subtracting half the total thickness gets back to the centre, for a square centred on the belt. It looks spot on to me.

image

Comment (martinbudden) [reply vitamins/belt.scad:67]: OK, that makes sense. I had a misunderstanding of the pitch height.

Comment (nophead) [reply vitamins/belt.scad]: So how is the user specified y gap centred? Or do you pass in a length twice the actual gap?

Comment (nophead) [reply vitamins/belt.scad]: I think the correct y gap needs to be passed in because it also needs to be subtracted from the length.

Comment (martinbudden) [reply vitamins/belt.scad]: The y cap is a static property of the belts layout. It is the belt separation in the y axis plus the belt thickness. It certainly needs to be passed in as a parameter.

Comment (nophead) [reply vitamins/belt.scad]: So if you pass the actual y gap how is the square to remove it working without adding half the gap to centre it correctly?

And gap.y also needs to be subtracted from the length added to the BOM.

Comment (martinbudden) [reply vitamins/belt.scad]: Not sure what you mean about centering the gap, the call is:
```
square(is_list(gap) ? [gap.x, gap.y] : [gap, thickness + eps], center = true);
```
that is the center parameter is set to true.

As for the length of the belts, yes it needs to be subtracted, I forgot that.

Comment (nophead) [reply vitamins/belt.scad]: Yes I guess you just pass the centre of the gap in the ```gap_pos```. 

Does the y value you pass in need to be inflated by the belt width as it is from pitch line to pitch line? 

So I think you subtract gap.x + gap.y from the length but need a square gap.y + thickness + eps to remove it.

Comment (martinbudden) [reply vitamins/belt.scad]: Yes, the y value needs to be inflated by the thickness. In my code I pass in the the `ySeparation+2` just to be sure, but that would result in an incorrect belt length. So I think the parameter passed in should be just the separation (not inflated by the thickness) and the square should add the thickness, ie:
```
square(is_list(gap) ? [gap.x, gap.y + thickness + eps] : [gap, thickness + eps], center = true);
```

Comment (nophead) [reply vitamins/belt.scad]: Yes I think that is the way to go.

**Comment by martinbudden** [comment]
I've fixed the belt length, so this should be ready to go.

**Pull request merged**